### PR TITLE
Use `ChatVectorDBChain` for integrating history.

### PR DESCRIPTION
**Pull request by h-mayorquin**
Here is a draft for how to do this:

image

I am wondering how does this impact the cost. How could we build tests for this?

### CODE REVIEW

File: src/spikeinterface_chatbot/app.py [hunk 1]
```diff
@@ -10,9 +10,11 @@
 @app.route("/", methods=["GET", "POST"])
 def chat():
     if request.method == "POST":
+        chat_history = [] if not locals().get("chat_history") else chat_history  # Hack to initalize, TODO: Improve
         message = request.form["message"]
         persist_directory = "./data/chroma_database"
-        response = query_documentation(query=message, persist_directory=persist_directory)
+        response = query_documentation(query=message, chat_history=chat_history, persist_directory=persist_directory)
+        chat_history.append(message)
```
Comment (bendichter) [inline src/spikeinterface_chatbot/app.py]: Is the response also added to the chat history or just the user messages?

Comment (h-mayorquin) [reply src/spikeinterface_chatbot/app.py]: It was adding only the questions. I am adding the two now in the last commit.

I implemented a queque as unbounded history will surpass the API token limit and crash.

**Comment by h-mayorquin** [comment]
Maybe we could use their buffers instead of a queue:

https://langchain.readthedocs.io/en/latest/modules/memory/types/buffer.html

**Comment by h-mayorquin** [comment]
Closed, as this has diverged from #2 and #3 .

**Pull request closed without merging**